src/css3_parser/parser.rs

```diff
@@ -0,0 +1,500 @@
+use super::{
+    tokenizer::CSSTokenizer,
+    tokens::{Token, TokenType},
+};
+
+#[derive(Debug, PartialEq)]
+pub struct CSSStyleSheet {
+    css_rules: Vec<CSSRule>,
+}
+
+impl Default for CSSStyleSheet {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl CSSStyleSheet {
+    pub fn new() -> CSSStyleSheet {
+        CSSStyleSheet {
+            css_rules: Vec::new(),
+        }
+    }
+}
+
+pub type CSSPropName = String;
+pub type CSSPropValue = String;
+
+#[derive(Debug, PartialEq)]
+pub struct CSSStyleDeclaration {
+    name: CSSPropName,
+    value: CSSPropValue,
+}
+
+#[derive(Debug, PartialEq)]
+pub struct CSSSelector {
+    path: Vec<String>,
+}
+
+impl Default for CSSSelector {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl CSSSelector {
```

Comment:
I'm thinking it would eventually be good to drop the "CSS" prefix, and do something like
```rust
use crate::gosub::css3::{Selector, StyleRule};
// ...
```

But I wouldn't attempt that change in this PR, since it implies larger changes as well.